README.md

```diff
@@ -1,33 +1,164 @@
+
+````typescript
+interface Car {
+  make: string
+  horsepower: number
+}
+
+// data has type Car
+var data = await ApiClient.get<Car>('/listings/search')
+
+// data has type { id: string } and body is typed as Car
+var data = await ApiClient.post<{ id: string }, Car>('/listings/create',
+  { make: 'bmw', horsepower: 300 },)
+````
+
+### Parameters
+
+If your request URL contains parameters, you can use curly brackets and the ApiClient replaces it with the passed data.
+This is aligned with the OpenAPI specification used by Swagger. Hence, you can easily search for occurrences of your
+APIs.
+
+````typescript
+// fetch is called with https://baseUrl.api.ch/dealers/123/listings/456
+await ApiClient.get('dealers/{dealerId}/listings/{listingId}', {
+  params: { dealerId: 123, listingId: 456 },
+})
+````
+
+### Authorization
+
+You can pass `accessToken` to the request options if you are talking to a protected API. The client will automatically
+set the header with a Bearer token.
+
+````typescript
+// fetch is called with the header { Authorization: `Bearer ${accessToken}` }
+await ApiClient.post<{ id: number }, { name: string; listingIds: number[] }>(
+  "users/me/listing-comparisons",
+  { name: "test1", listingIds: [890163] },
+  { accessToken: authHeader.accessToken }
+)
+````
+
+### Error handling
+
+The promise is rejected if any error occurs on API level. Use `try/catch` to add error handling.
+
+````typescript
+try {
+  // res has the type LeasingCalculation
+  const res = await ApiClient.post<LeasingCalculation, LeasingData>(
+    "listings/calculate-leasing",
+    {
+      downPayment: 7300,
+      duration: 48,
+      estimatedKmPerYear: 10000,
+      firstRegistrationDate: "2020-07-01",
+      residualValue: 12045,
+      price: 36500,
+    }
+  )
+} catch (error: ResponseError) {
+  // do any error handling you want
+}
+````
+
+### Testing
+
+Create a file in the `__mocks__` directory called `@smg-automotive/api-client-pkg.ts` with the following content:
+
+````typescript
+export const ApiClient = {
+  get: jest.fn().mockReturnValue(Promise.resolve()),
+  post: jest.fn().mockReturnValue(Promise.resolve()),
+  put: jest.fn().mockReturnValue(Promise.resolve()),
+  delete: jest.fn().mockReturnValue(Promise.resolve()),
+}
+export const ResponseError = Error
+````
+
+By default, all the API calls will succeed. If you want to reject it or return a specific value you can do this as
+follows:
+
+````typescript
+jest.spyOn(ApiClient, "get").mockImplementation((path, options) => {
+  // do something with path and options (e.g. create a switch statement for the methods used)
+  return Promise.resolve([{ name: "bmw", key: "bmw" }])
+})
+
+jest.spyOn(ApiClient, "get").mockImplementation((path, options) => {
+  // do something with path and options (e.g. create a switch statement for the methods used)
+  return Promise.reject(
+    new ResponseError({ status: 422, statusText: "" }, "Some data from the response body")
```

Comment:
where is `ResoponseError` defined? can you include the import statement in the example?